src/main/java/graphql/schema/idl/SchemaPrinterComparatorEnvironment.java

```diff
@@ -0,0 +1,93 @@
+package graphql.schema.idl;
+
+import graphql.Assert;
+import graphql.schema.GraphQLType;
+
+/**
+ * Defines the scope to control where the registered {@code Comparator} can be applied.
+ * <p></p>
+ * {@code elementType}s can be ordered within its {@code parentType} to restrict the {@code Comparator}s scope of operation.
+ * Otherwise supplying only the {@code elementType} results in the {@code Comparator} being reused across all matching {@code GraphQLType}s regardless of parent.
+ */
+public class SchemaPrinterComparatorEnvironment {
+
+    private Class<? extends GraphQLType> parentType;
+
+    private Class<? extends GraphQLType> elementType;
+
+    private SchemaPrinterComparatorEnvironment(Class<? extends GraphQLType> parentType, Class<? extends GraphQLType> elementType) {
+        Assert.assertNotNull(elementType, "elementType can't be null");
+        this.parentType = parentType;
+        this.elementType = elementType;
+    }
+
+    /**
+     * @return The parent type or {@code null} if not supplied.
+     */
+    public Class<? extends GraphQLType> getParentType() {
+        return parentType;
+    }
+
+    /**
+     * @return A new environment containing only the element type.
+     */
+    public SchemaPrinterComparatorEnvironment withElementTypeOnly() {
+        return new SchemaPrinterComparatorEnvironment(null, elementType);
+    }
```

Comment:
Typically we would do with the "transform"

Have a look for "transform(Consumer<Builder> builderFunction)" examples

it allows things like

```
newEnv = env.transform( builder -> builder.parentType(null));
```

See https://github.com/graphql-java/graphql-java/blob/master/src/main/java/graphql/GraphQL.java#L209 say